beetsplug/convert.py

```diff
@@ -399,15 +434,36 @@ def copy_album_art(self, album, dest_dir, path_formats, pretend=False):
             if not pretend:
                 ArtResizer.shared.resize(maxwidth, album.artpath, dest)
         else:
-            if pretend:
-                self._log.info(u'cp {0} {1}',
-                               util.displayable_path(album.artpath),
-                               util.displayable_path(dest))
+            if hardlink:
+                if pretend:
+                    self._log.info(u'ln {0} {1}',
+                                   util.displayable_path(album.artpath),
+                                   util.displayable_path(dest))
+                else:
+                    self._log.info(u'Hardlinking cover art from {0} to {1}',
+                                   util.displayable_path(album.artpath),
+                                   util.displayable_path(dest))
+                    util.hardlink(album.artpath, dest)
+            elif link:
+                if pretend:
+                    self._log.info(u'ln -s {0} {1}',
+                                   util.displayable_path(album.artpath),
+                                   util.displayable_path(dest))
+                else:
+                    self._log.info(u'Linking cover art from {0} to {1}',
+                                   util.displayable_path(album.artpath),
+                                   util.displayable_path(dest))
+                    util.link(album.artpath, dest)
```

Comment:
As above—it would be great to reduce the duplicated code between these branches.